packages/contracts-bedrock/src/L1/OptimismPortal.sol

```diff
@@ -398,6 +409,15 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
         // Emit a TransactionDeposited event so that the rollup node can derive a deposit
         // transaction for this deposit.
         emit TransactionDeposited(from, _to, DEPOSIT_VERSION, opaqueData);
+
+        // Compute the updated global hash union of deposit data and cache the previous value on the stack for
+        // event emission.
+        bytes32 unionBefore = depositHashUnion;
+        bytes32 unionAfter = keccak256(abi.encode(unionBefore, from, _to, DEPOSIT_VERSION, opaqueData));
+
+        // Persist the updated global hash union of deposit data and emit a HashUnionEvent update.
+        depositHashUnion = unionAfter;
+        emit HashUnionUpdated(unionBefore, unionAfter);
```

Comment:
This feels like tech debt adding another event, we should append the extra data to `opaqueData` and then increment the `DEPOSIT_VERSION`. We will be able to use most of the same derivation pipeline parsing code

A question that I have is that this change seems sort of abrupt, was this an expected thing that was known for a long time? Or did this come from a change in the 4844 workstream?